spock-spring/src/main/java/org/spockframework/spring/SpringMockTestExecutionListener.java

```diff
@@ -61,6 +64,10 @@ public void beforeTestMethod(TestContext testContext) throws Exception {
     }
   }
 
+  private boolean scanScopedBean(ScanScopedBeans scanScopedBeans, Set<String> scopes, BeanDefinition beanDefinition) {
+    return scanScopedBeans != null && (scopes.size() == 0 || scopes.contains(beanDefinition.getScope()));
```

Comment:
You already have `Set<String> scopes = scanScopedBeans == null ? Collections.<String>emptySet() : new HashSet<String>(Arrays.asList(scanScopedBeans.value()));` so you just need `scopes.contains(beanDefinition.getScope()`; when disabled `scopes` is empty otherwise it's set from the annotation.